arrow-fx-coroutines/src/test/kotlin/arrow/fx/coroutines/predef-test.kt

```diff
@@ -203,14 +203,16 @@ internal fun <A> Either<Throwable, A>.suspended(): suspend () -> A =
  * ```
  * @see Assertions.assertThrows
  */
-inline fun <A> assertThrowable(executable: () -> A): Throwable =
-  try {
-    val a = executable.invoke()
-    fail("Expected an exception but found: $a")
+inline fun <A> assertThrowable(executable: () -> A): Throwable {
```

Comment:
I discovered that we can call `shouldThrow` instead, do we need this function at all?

Reply:
If it offers the same behavior then we definitely don't need this one!
I replaced this instead of an `inline reified` one since I'm always just capturing `Throwable` and then use `shouldBe` instead.

So I use it the same as `Either.catch { } shouldBe Either.Left(e)` vs `assertThrowable { } shouldBe e`